react-app-drawer/my-app/src/components/AppDrawer.tsx

```diff
@@ -0,0 +1,56 @@
+import { useState } from 'react';
+import { Link, Outlet } from 'react-router-dom';
+
+export type MenuItem = {
+  name: string;
+  iconUrl: string;
+  path: string;
+};
+
+type Props = {
+  menuItems: MenuItem[];
+};
+export function AppDrawer({ menuItems }: Props) {
+  const [isOpen, setIsOpen] = useState(false);
+  let navWidth = 'navbar-open';
+  if (isOpen) {
+    navWidth = 'navbar-open';
+  } else {
+    navWidth = 'navbar-close';
+  }
```

Comment:
We would normally write the above more succinctly as:
`const navWidth = isOpen ? 'navbar-open' : 'navbar-close';`